File: python/problems/lc14/solution.py
"""
[LC] 14 - Longest Common Prefix
https://leetcode.com/problems/longest-common-prefix/
난이도: easy
태그: array

LeetCode 제출용 시그니처를 직접 작성하세요.
"""
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]

        strs.sort()
        first = strs[0]
        last = strs[-1]
        answer = ""
        for i in range(len(first)):
            if first[i] == last[i]:
                answer += first[i]
            else:
                break
        return answer

File: go/problems/lc14/solution.go
// Package lc14 - [LC] 14 - Longest Common Prefix
// https://leetcode.com/problems/longest-common-prefix/
// 난이도: easy
// 태그: array
//
// LeetCode 제출용 시그니처를 직접 작성하세요.
package lc14

import "sort"

func longestCommonPrefix(strs []string) string {
	if len(strs) == 1 {
		return strs[0]
	}
	sort.Strings(strs)

	first := strs[0]
	last := strs[len(strs)-1]
	for i := 0; i < len(first); i++ {
		if first[i] != last[i] {
			return first[:i]
		}
	}
	return first
}
